### PR TITLE
Added timeoutafter to DotNetty eventloop calls

### PR DESCRIPTION
**Pull request by brotherneeru**
Cherrypick of https://github.com/Azure/iotedge/pull/6659

### CODE REVIEW

**Review by ancaantochi — COMMENTED** [review]
why was the timeout for protocol heads was not ported from 1.1?

**Comment by brotherneeru** [comment]
@ancaantochi Good catch! I have that on my local Ill merge it in

File: edge-hub/core/src/Microsoft.Azure.Devices.Edge.Hub.Service/Program.cs [hunk 1]
```diff
@@ -121,7 +121,9 @@ static async Task<int> MainAsync(IConfigurationRoot configuration)
                 }
 
                 TimeSpan shutdownWaitPeriod = TimeSpan.FromSeconds(configuration.GetValue("ShutdownWaitPeriod", DefaultShutdownWaitPeriod));
-                (CancellationTokenSource cts, ManualResetEventSlim completed, Option<object> handler) = ShutdownHandler.Init(shutdownWaitPeriod, logger);
+                (CancellationTokenSource cts, ManualResetEventSlim completed, Option<object> handler) = ShutdownHandler.Init(shutdownWaitPeriod, logger);
+                var protocolTimeout = configuration.GetValue("protocolTimeoutInSecs", 180);
+                TimeSpan protocolTimeoutinSecs = TimeSpan.FromMilliseconds(protocolTimeout);
```
Comment (ancaantochi) [inline edge-hub/core/src/Microsoft.Azure.Devices.Edge.Hub.Service/Program.cs:126]: should use FromSeconds instead?

Comment (brotherneeru) [reply edge-hub/core/src/Microsoft.Azure.Devices.Edge.Hub.Service/Program.cs]: Yup, thanks for catching

**Comment by damonbarry** [comment]
@brotherneeru Did you test this change in 1.4?

**Review by ancaantochi — APPROVED** [review]

**Pull request merged**